toolkits/global/packages/global-article/js/show-more.js

```diff
@@ -0,0 +1,116 @@
+/* global jQuery $ */
+
+function ShowMore() {
+	'use strict';
+
+	var settings;
+
+	var init = function (s, scheduler, emitter) {
+		settings = s;
+
+		settings.$el.each(function () {
+			var $element = $(this);
+			var $button;
+			var fullText = $element.text().trim();
+			var fullHtml = $element.html();
+			var hasButton = !$element.attr('data-hellip-disable-expand');
+			var hasShowLess = !$element.attr('data-hellip-disable-show-less');
+			var truncatedText;
+
+			function insertButton($button, $element, buttonText) {
+				var $inner = $element.children(':last-child').filter(function (i, element) {
+					return $(element).css('display') === 'block';
+				});
+				// if the last child is block level we'll append the button there
+				if ($inner.length === 0) {
+					$inner = $element;
+				}
+				$inner.append($button);
+				$button.html(buttonText).before(' ');
+			}
+
+			// eslint-disable-next-line unicorn/consistent-function-scoping
+			function truncateText(html, length) {
+				return $.truncate(html, {
+					length: length,
+					words: true
+				});
+			}
+
+			function setUp(html, previewLength) {
+				if (fullText.length >= previewLength) {
+					truncatedText = truncateText(fullHtml, previewLength);
+					$element.html(truncatedText);
+					if (hasButton) {
+						$button = $('<button class="link-like inline"' + settings.linkAttribute + '="" >' + settings.showMoreText + '</button>');
+						if (settings.trackAttribute) {
+							$button.attr('data-track', 'click');
+							$button.attr('data-track-label', 'button');
+							$button.attr('data-track-action', settings.trackAttribute + ' show less');
+						}
+						insertButton($button, $element, settings.showMoreText);
+					}
+				}
+			}
+
+			function setTrackingFor($button, state) {
+				if (settings.trackAttribute) {
+					$button.attr('data-track-action', settings.trackAttribute + ' show ' + state);
+				}
+			}
+
+			if (hasButton) {
+				$element.on('click', 'button[' + settings.linkAttribute + ']', function () {
+					if ($element.hasData(settings.expandedAttr)) {
```

Comment:
This is the line that is giving me the `TypeError: $element.hasData is not a function` when running the `Should toggle button between show more and show less` test